mlir/lib/Analysis/FlatLinearValueConstraints.cpp

```diff
@@ -86,7 +86,8 @@ getFlattenedAffineExprs(ArrayRef<AffineExpr> exprs, unsigned numDims,
     if (!expr.isPureAffine())
       return failure();
 
-    flattener.walkPostOrder(expr);
+    auto flattenResult = flattener.walkPostOrder(expr);
+    assert(succeeded(flattenResult) && "affine expr containts poison expr");
```

Comment:
this will cause unused variable warnings in non-debug builds
```suggestion
    (void)flattenResult;
    assert(succeeded(flattenResult) && "affine expr containts poison expr");
```